### PR TITLE
Adding recipe for exprtk

### PR DESCRIPTION
**Pull request by vijaykhemka**
The C++ Mathematical Expression  Toolkit Library (ExprTk) is  a simple
to  use,   easy  to   integrate  and   extremely  efficient   run-time
mathematical  expression parsing  and evaluation  engine. The  parsing
engine  supports numerous  forms  of  functional and  logic processing
semantics and is easily extensible.

It is a header only library.

Signed-off-by: Vijay Khemka <vijaykhemka@fb.com>

### CODE REVIEW

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+SUMMARY = "Expression parser"
+HOMEPAGE = "https://github.com/ArashPartow/exprtk"
+SECTION = "libs"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COREBASE}/meta/files/common-licenses/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
```
Comment (kraj) [inline meta-oe/recipes-devtools/exprtk/exprtk_git.bb:5]: ${COREBASE}/LICENSE is not a valid license file, please use
'${COMMON_LICENSE_DIR}/MIT'

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+RDEPENDS_${PN}-dev = ""
+
+BBCLASSEXTEND = "native nativesdk"
+
+# other packages commonly reference the file directly as "exprtk.hpp"
+# create symlink to allow this usage
+do_install() {
+		install -d ${D}/${includedir}
+    install -m 0755 ${S}/exprtk.hpp ${D}/${includedir}/exprtk.hpp
```
Comment (kraj) [inline meta-oe/recipes-devtools/exprtk/exprtk_git.bb:21]: mode should be 0644

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# exprtk is a header only C++ library, so the main package will be empty.
+
+RDEPENDS_${PN}-dev = ""
+
+BBCLASSEXTEND = "native nativesdk"
+
+# other packages commonly reference the file directly as "exprtk.hpp"
+# create symlink to allow this usage
+do_install() {
+		install -d ${D}/${includedir}
```
Comment (kraj) [inline meta-oe/recipes-devtools/exprtk/exprtk_git.bb:20]: adjust formatting please

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COREBASE}/meta/files/common-licenses/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+SRC_URI = "git://github.com/ArashPartow/exprtk.git;nobranch=1"
+SRCREV = "281c2ccc65b8f91c012ea3725ebcef406378a225"
+S = "${WORKDIR}/git"
+
+# exprtk is a header only C++ library, so the main package will be empty.
+
+RDEPENDS_${PN}-dev = ""
```
Comment (kraj) [inline meta-oe/recipes-devtools/exprtk/exprtk_git.bb:13]: run it through meta-openembedded/contrib/oe-stylize.py script so sections are arranged correctly.

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+SUMMARY = "Expression parser"
+HOMEPAGE = "https://github.com/ArashPartow/exprtk"
+SECTION = "libs"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COREBASE}/meta/files/common-licenses/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+SRC_URI = "git://github.com/ArashPartow/exprtk.git;nobranch=1"
```
Comment (kraj) [inline meta-oe/recipes-devtools/exprtk/exprtk_git.bb:7]: why is nobranch needed ?

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb (hunk 1 above)
Comment (vijaykhemka) [reply meta-oe/recipes-devtools/exprtk/exprtk_git.bb:5]: Sure, I will update this.

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb (hunk 2 above)
Comment (vijaykhemka) [reply meta-oe/recipes-devtools/exprtk/exprtk_git.bb:21]: Ack

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb (hunk 3 above)
Comment (vijaykhemka) [reply meta-oe/recipes-devtools/exprtk/exprtk_git.bb:20]: Ack

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb (hunk 4 above)
Comment (vijaykhemka) [reply meta-oe/recipes-devtools/exprtk/exprtk_git.bb:13]: Ack

File: meta-oe/recipes-devtools/exprtk/exprtk_git.bb (hunk 5 above)
Comment (vijaykhemka) [reply meta-oe/recipes-devtools/exprtk/exprtk_git.bb:7]: Wanted to take master branch only, I will remove this.

**Comment by vijaykhemka** [comment]
Khem, Please review and let me know if I need to make further modification. I need to use this tool in one of our project.

**Comment by kraj** [comment]
> Khem, Please review and let me know if I need to make further modification. I need to use this tool in one of our project.

its on master-next already, so far no issues testing will complete soon

**Comment by kraj** [comment]
merged with dd0fafdc53834c9487dc292ea7a2d870c9654e55

**Pull request closed without merging**